kore/src/Kore/Step/Simplification/Not.hs

```diff
@@ -174,11 +174,25 @@ makeTermNot
 -- TODO: maybe other simplifications like
 -- not ceil = floor not
 -- not forall = exists not
+makeTermNot (Not_ _ term) = term
+makeTermNot (And_ _ term1 term2) =
+    mkOr (makeTermNot term1) (makeTermNot term2)
 makeTermNot term
   | isBottom term = mkTop_
   | isTop term    = mkBottom_
   | otherwise     = mkNot term
 
+makePredicateNot
+    :: InternalVariable variable
+    => Syntax.Predicate.Predicate variable
+    -> Syntax.Predicate.Predicate variable
+makePredicateNot Syntax.Predicate.PredicateFalse =
+    Syntax.Predicate.makeTruePredicate
+makePredicateNot Syntax.Predicate.PredicateTrue =
+    Syntax.Predicate.makeFalsePredicate
+makePredicateNot genericPredicate =
+    fmap makeTermNot genericPredicate
```

Comment:
See my comment above, this fmap always produces a `Syntax.Predicate`, although that shouldn't always be the case.